bundles/org.openhab.binding.solarman/src/main/resources/OH-INF/thing/channels.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="solarman"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-type id="string">
+		<item-type>String</item-type>
+		<label>Text Value</label>
+		<config-description-ref uri="channel-type-config:solarman:string_channel"/>
+	</channel-type>
+
+	<channel-type id="number">
+		<item-type>Number</item-type>
```

Comment:
Not sure where all numbers are used for, but would be nice if somehow this can be adapted to channels that support Units of Measure

Reply:
These are used for user defined custom requests, for example:
```
        Channels:
        Type number : Inverter_Frequency [scale="0.01", uom="Hz", rule="3", registers="0x27E"]
```
and based on that definition a UoM is set on the item. Here is a screen shot.

![image](https://github.com/openhab/openhab-addons/assets/25250295/91fd2b14-2206-4dd5-ac19-dc8fa77e026b)

But I think you mean something different, probably along https://www.openhab.org/docs/concepts/units-of-measurement.html#defining-channeltypes , but because those are custom defined I don't know ahead of time the Item type.